src/views/Document.vue

```diff
@@ -41,13 +38,19 @@
 import { mapGetters } from 'vuex';
 import ResultHeader from '../components/Result/ResultHeader';
 import Toggle from '@/components/Global/Toggle/Toggle';
+import { prevRoute } from '../router/index.js';
 
 export default {
 	components: {
 		ResultHeader,
 		Toggle
 	},
 	computed: {
+		backToPrev: function () {
+			if (prevRoute.name === null)
```

Comment:
Just thinking. Using named routes in totally optional in Vue. You don't have to specify a `name` for route. In that case, it would redirect to result page, even if the previous page was from our domain, which is weird. 

What I noticed by running the code, is that fullPath is always `/` when coming from elsewhere (accessing our router for the first time).

![image](https://user-images.githubusercontent.com/27869979/113504697-02a5e380-953a-11eb-81a3-f802e763ab11.png)


![image](https://user-images.githubusercontent.com/27869979/113504727-35e87280-953a-11eb-9d72-220c9f09674e.png)

So maybe `fullPath === '/' && name == null`? 

```
export let prevRoute = undefined;

router.beforeEach((to, from, next) => {
    // do not cache prevRoute on first access to app
    if (from.fullPath !== '/' && from.name !== null) {
         prevRoute = from;
    }
	next();
});
```
And then in component:
```
if (prevRoute === undefined) {
   return result
}
return prevRoute
```

What do you think of that?